### PR TITLE
Set vlan_mac and loopback ip for dual tor

### PR DESCRIPTION
**Pull request by kellyyeh**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205

#### What is the motivation for this PR?
Dual tor has different mac addresses for router mac and vlan mac. Need to specify the correct mac address for server->t1 packets to be received successfully at the DUT
Dual tor device also has different loopback address. Updated test to use the correct one.

#### How did you do it?
Update vlan mac and lo prefix for dual tor test

#### How did you verify/test it?
Ran test manually on dual tor device, verify server->t1 packets are received

### CODE REVIEW

File: ansible/roles/test/files/ptftests/advanced-reboot.py [hunk 1]
```diff
@@ -747,17 +751,17 @@ def generate_from_vlan(self):
...
     def generate_ping_dut_lo(self):
         self.ping_dut_packets = []
-        dut_lo_ipv4 = self.test_params['lo_prefix'].split('/')[0]
+        dut_lo_ipv4 = self.lo_prefix.split('/')[0]
         for src_port in self.vlan_host_ping_map:
             src_addr = random.choice(self.vlan_host_ping_map[src_port].keys())
             src_mac = self.hex_to_mac(self.vlan_host_ping_map[src_port][src_addr])
-            packet = simple_icmp_packet(eth_src=src_mac,
-                                        eth_dst=self.dut_mac,
+            packet = simple_icmp_packet(eth_src=self.from_server_src_mac,
```
Comment (vaibhavhd) [inline ansible/roles/test/files/ptftests/advanced-reboot.py:758]: Why is this change needed for src_mac? src_mac in this change becomes same for all the servers, but that's not true right?

File: ansible/roles/test/files/ptftests/advanced-reboot.py [hunk 2]
```diff
@@ -747,17 +751,17 @@ def generate_from_vlan(self):
...
+                                        eth_dst=self.vlan_mac,
```
Comment (vaibhavhd) [inline ansible/roles/test/files/ptftests/advanced-reboot.py:759]: Although this will work for dualtor device, I think it will break single TOR. At the beginning of the test, you can decide what will be the eth_dst for server->t1 traffic.
Then use that variable everywhere?

File: ansible/roles/test/files/ptftests/advanced-reboot.py [hunk 3]
```diff
@@ -747,17 +751,17 @@ def generate_from_vlan(self):
...
                                         ip_src=src_addr,
                                         ip_dst=dut_lo_ipv4)
             self.ping_dut_packets.append((src_port, str(packet)))
 
-        exp_packet = simple_icmp_packet(eth_src=self.dut_mac,
+        exp_packet = simple_icmp_packet(eth_src=self.vlan_mac,
```
Comment (vaibhavhd) [inline ansible/roles/test/files/ptftests/advanced-reboot.py:764]: Same comment -- works for dualtor, breaks single tor.

File: ansible/roles/test/files/ptftests/advanced-reboot.py [hunk 4]
```diff
@@ -766,6 +770,7 @@ def generate_ping_dut_lo(self):
                                     ip_dst=dut_lo_ipv4)
 
         self.ping_dut_exp_packet  = Mask(exp_packet)
+        self.ping_dut_exp_packet.set_do_not_care_scapy(scapy.Ether, "src")
```
Comment (vaibhavhd) [inline ansible/roles/test/files/ptftests/advanced-reboot.py:773]: I think you don't need this, once you remove the change for src_mac from above.

File: tests/common/fixtures/advanced_reboot.py [hunk 5]
```diff
@@ -168,7 +168,17 @@ def __buildTestbedData(self, tbinfo):
...
 
         self.rebootData['dut_hostname'] = self.mgFacts['minigraph_mgmt_interface']['addr']
+        config_facts = self.duthost.get_running_config_facts()
+        vlan_table = config_facts['VLAN']
+        vlan_name = list(vlan_table.keys())[0]
+        vlan_mac = vlan_table[vlan_name]['mac']
+        self.rebootData['vlan_mac'] = vlan_mac
         self.rebootData['dut_mac'] = self.duthost.facts['router_mac']
+        if 'dualtor' in tbinfo['topo']['name']:
+            self.rebootData['lo_prefix'] = "10.1.0.33/32"
```
Comment (vaibhavhd) [inline tests/common/fixtures/advanced_reboot.py:178]: Can we get this data from duthost.facts or config facts, instead of hardcoding?

File: ansible/roles/test/files/ptftests/advanced-reboot.py (hunk 2 above)
Comment (vaibhavhd) [reply ansible/roles/test/files/ptftests/advanced-reboot.py:759]: Correction -- vlan_mac and router_mac are same for single tor. This should work.

File: ansible/roles/test/files/ptftests/advanced-reboot.py (hunk 4 above)
Comment (kellyyeh) [reply ansible/roles/test/files/ptftests/advanced-reboot.py]: Removed

File: ansible/roles/test/files/ptftests/advanced-reboot.py (hunk 1 above)
Comment (kellyyeh) [reply ansible/roles/test/files/ptftests/advanced-reboot.py]: Correct this change is not needed, fixed!

File: tests/common/fixtures/advanced_reboot.py (hunk 5 above)
Comment (kellyyeh) [reply tests/common/fixtures/advanced_reboot.py]: Updated

File: ansible/roles/test/files/ptftests/advanced-reboot.py [hunk 6]
```diff
@@ -747,17 +751,17 @@ def generate_from_vlan(self):
...
     def generate_ping_dut_lo(self):
         self.ping_dut_packets = []
-        dut_lo_ipv4 = self.test_params['lo_prefix'].split('/')[0]
+        dut_lo_ipv4 = self.lo_prefix.split('/')[0]
         for src_port in self.vlan_host_ping_map:
             src_addr = random.choice(self.vlan_host_ping_map[src_port].keys())
             src_mac = self.hex_to_mac(self.vlan_host_ping_map[src_port][src_addr])
-            packet = simple_icmp_packet(eth_src=src_mac,
-                                        eth_dst=self.dut_mac,
+            packet = simple_icmp_packet(eth_src=src_mac,,
```
Comment (vaibhavhd) [inline ansible/roles/test/files/ptftests/advanced-reboot.py:758]: Extra `,` added. Surprising how this did not fail the call to `simple_icmp_packet`.

Comment (kellyyeh) [reply ansible/roles/test/files/ptftests/advanced-reboot.py]: Was supposed to be added to latest commit but missed, fixed.

File: tests/common/fixtures/advanced_reboot.py [hunk 7]
```diff
@@ -168,7 +168,14 @@ def __buildTestbedData(self, tbinfo):
         self.vlanMaxCnt = len(self.mgFacts['minigraph_vlans'].values()[0]['members']) - 1
 
         self.rebootData['dut_hostname'] = self.mgFacts['minigraph_mgmt_interface']['addr']
+        config_facts = self.duthost.get_running_config_facts()
+        vlan_table = config_facts['VLAN']
+        vlan_name = list(vlan_table.keys())[0]
```
Comment (yxieca) [inline tests/common/fixtures/advanced_reboot.py:173]: Will vlan name always be there?

Comment (kellyyeh) [reply tests/common/fixtures/advanced_reboot.py]: Changed single tor to keep use router mac, as for dual tor seems like at least one vlan will always be there

File: tests/common/fixtures/advanced_reboot.py [hunk 8]
```diff
@@ -169,6 +169,16 @@ def __buildTestbedData(self, tbinfo):
...
         self.rebootData['dut_hostname'] = self.mgFacts['minigraph_mgmt_interface']['addr']
         self.rebootData['dut_mac'] = self.duthost.facts['router_mac']
+        if "dualtor" in tbinfo['topo']['name']:
+            config_facts = self.duthost.get_running_config_facts()
+            vlan_table = config_facts['VLAN']
+            vlan_name = list(vlan_table.keys())[0]
+            vlan_mac = vlan_table[vlan_name]['mac']
+            self.rebootData['vlan_mac'] = vlan_mac
+        else:
+            self.rebootData['vlan_mac'] = self.rebootData['dut_mac']
```
Comment (yxieca) [inline tests/common/fixtures/advanced_reboot.py:179]:
```suggestion
        vlan_mac = self.rebootData['dut_mac']
        config_facts = self.duthost.get_running_config_facts()
        vlan_table = config_facts.get('VLAN', None)
        if vlan_table:
            vlan_name = list(vlan_table.keys())[0]
            vlan_mac = vlan_table[vlan_name].get('mac'. self.rebootData['dut_mac'])
        self.rebootData['vlan_mac'] = vlan_mac
```

**Review by vaibhavhd — APPROVED** [review]
